src/cmd/poll/actor.rs

```diff
@@ -210,4 +220,38 @@ impl Actor<PollMessage> for PollActor {
       }
     }
   }
+
+  async fn shutdown(&mut self) -> Result<()> {
+    let poll_count = self.states.len().unwrap_or(0);
+    info!(
+      "Shutting down PollActor - saving {} active polls",
+      poll_count
+    );
+
+    // Get all active poll IDs and save each one to persistence
+    match self.states.keys() {
```

Comment:
Why was `iter` impld if this is just going keys -> invoke -> save? Maybe we can just iterate? Nesting on this block is also really bad/needs cleanup